Search result component

kasperbirch1 opened the pull request:
A component with an inline heart svg. We have done this to be able to receive a prop that determines whether the heart should be fill.

kasperbirch1 committed:
Heart toggle icon

A component with an inline heart svg. We have done this to be able to receive a prop that determines whether the heart should be fill.

kasperg commented on src/stories/icon/icon-heart/IconHeart.stories.tsx:
```diff
@@ -0,0 +1,32 @@
+    design: {
+      type: "figma",
+      url:
+        "https://www.figma.com/file/Zx9GrkFA3l4ISvyZD2q0Qi/Designsystem?node-id=1385%3A10755",
```
I do not think this is the most precise link for the component. If I go to Figma, locate the wishlist icon in the left hand side, click "Copy link" I get https://www.figma.com/file/Zx9GrkFA3l4ISvyZD2q0Qi/Designsystem?node-id=656%3A5407

kasperg commented on src/stories/icon/icon-heart/IconHeart.stories.tsx:
```diff
@@ -0,0 +1,32 @@
+);
+
+export const Heart = Template.bind({});
+Heart.args = {
```
I am unsure about the naming here. The README states that we use the same naming convention here as in Figma. Figma uses the word "Wishlist" but that does not apply very well. Another option might be "Favourite" which applies better.

kasperg commented on src/stories/icon/icon-heart/IconHeart.tsx:
```diff
@@ -0,0 +1,21 @@
+      <path
+        d="M11.5 20L10.2675 18.921C5.89 15.1035 3 12.5858 3 9.49591C3 6.9782 5.057 5 7.675 5C9.154 5 10.5735 5.66213 11.5 6.70845C12.4265 5.66213 13.846 5 15.325 5C17.943 5 20 6.9782 20 9.49591C20 12.5858 17.11 15.1035 12.7325 18.9292L11.5 20Z"
+        stroke="black"
+        stroke-width="2"
```
PHPStorm reports that `stroke-width` is not allowed here and suggests `strokeWidth`. Does this ring a bell?

kasperg commented on src/stories/icon/icon-heart/IconHeart.tsx:
```diff
@@ -0,0 +1,21 @@
+
+export const IconHeart = ({ fill }: IconHeartProps) => {
+  return (
+    <svg
+      width="24"
+      height="24"
+      viewBox="0 0 24 24"
+      fill={fill ? "#000" : "none"}
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <path
+        d="M11.5 20L10.2675 18.921C5.89 15.1035 3 12.5858 3 9.49591C3 6.9782 5.057 5 7.675 5C9.154 5 10.5735 5.66213 11.5 6.70845C12.4265 5.66213 13.846 5 15.325 5C17.943 5 20 6.9782 20 9.49591C20 12.5858 17.11 15.1035 12.7325 18.9292L11.5 20Z"
```
As far as I can tell this is copy/pasted from `public/icons/basic/icon-heart.svg`. Please add a reference to this as a code comment referencing this to help future developers understand why this might suddenly deviate if the "official" icon is updated.

kasperbirch1 committed:
refactor component IconHeart / IconFavourite

refactor component name + and resolve all comments from previous pull request.

kasperbirch1 committed:
Add SearchResultElement component.

This is a component that displays results items. The component use these atoms
- Material
- IconFavourite
- AvailabilityLabel

kasperbirch1 committed:
Add SearchResultElement component.

This is a component that displays results items. The component use these atoms
- Material
- IconFavourite
- AvailabilityLabel

kasperbirch1 committed:
add controls and defaultValues to storybook argTypes.

kasperbirch1 committed:
add controls and defaultValues to storybook argTypes.

kasperg commented on base.scss:
```diff
@@ -26,6 +26,7 @@
 @import "./src/stories/counter/counter";
 @import "./src/stories/checkbox/checkbox";
 @import "./src/stories/availability-label/availability-label";
+@import "./src/stories/search-result-element/search-result-element.scss";
```
The other imports do not include the `.scss` suffix. Why add it here?

kasperg commented on src/stories/search-result-element/SearchResultElement.tsx:
```diff
@@ -0,0 +1,74 @@
+          manifestation={"Ebog"}
+          availability={"Online"}
+          status={"selected"}
+        />
```
Please remove the selected variant. It is not relevant in this context.

kasperg commented on src/stories/search-result-element/SearchResultElement.tsx:
```diff
@@ -0,0 +1,74 @@
+          animate={false}
+          tint={"120"}
+        />
+        <div className="search-result-element__info__text-content">
```
We need to break this into smaller blocks e.g. `search-result-element-text`.

kasperg replied:
Remember to add margin between text and availability elements.

kasperbirch1 replied:
I have used this name search-result-element-content because it is a container that contains both text and accessibility

kasperg commented on src/stories/search-result-element/SearchResultElement.tsx:
```diff
@@ -0,0 +1,74 @@
+          <div className="search-result-element__info__text-content__top">
+            <IconFavourite fill={heartFill} />
+
+            {categoryScore && (
```
`categoryScore` seems like a poor choice for a variable name. How about `seriesNumber`?

kasperg commented on src/stories/search-result-element/SearchResultElement.tsx:
```diff
@@ -0,0 +1,74 @@
+                    i serien{" "}
+                    <span className="search-result-element__info__text-content__top__category-score__underline">
+                      {series}
+                    </span>
```
This will normally be a link. I guess it will take the patron to a search result containing all parts of the series.

kasperg replied:
Look into the `link-tag` class to get the proper spacing between text and underline.

kasperg commented on src/stories/search-result-element/SearchResultElement.tsx:
```diff
@@ -0,0 +1,74 @@
+  series,
+}: SearchResultElementProps) => {
+  return (
+    <div className="search-result-element">
```
I wonder if/how element should link to the page displaying the material. Is it the title and cover image, the entire element or something else. We probably need to check with the design (and perhaps the designers).

Adamik10 replied:
Great point. There is at least a little something in Figma that addresses this:
![image](https://user-images.githubusercontent.com/28546954/171375304-1fd6e318-c1e2-470e-814c-b9a222c322ec.png)
My bet would be that the whole card is clickable, but let's talk to Dept about this. :) @kasperbirch1 could you add it to the agenda pls when you're resolving this?

kasperg commented on src/stories/search-result-element/search-result-element.scss:
```diff
@@ -0,0 +1,44 @@
+        align-items: center;
+
+        svg {
+          cursor: pointer;
```
Please move this to the icon favourite atom.

kasperg commented on src/stories/search-result-element/SearchResultElement.tsx:
```diff
@@ -0,0 +1,74 @@
+            )}
+          </div>
+          <div className="search-result-element__info__text-content__bottom">
+            <h3 className="text-header-h3">{title}</h3>
```
The titles look a bit big. It seems as if this should be styled with a h4.

kasperg replied:
Please also bump to a h2 as this seems to correspond with the document structure of search results.

kasperbirch1 committed:
Add css and renamed the folder.

kasperbirch1 committed:
Merge branch 'heart-toggle' into search-result-components

because it is an atom used in the SearchResultElement

kasperbirch1 committed:
switch from categoryScore to seriesNumber

because it is more descriptive.

kasperbirch1 committed:
removed the selected type

because it is not relevant in that context.

kasperbirch1 committed:
change to responsive

htlm markup and css have been modified to make the component responsive

kasperbirch1 committed:
pull and fix conflicts

kasperbirch1 committed:
Refactor css for more responsive layout

kasperg committed:
Refactor search result markup and CSS to reduce nesting

Use grid display to achieve a layout that switches between two 
columns on mobile (cover and text+availability) to three (cover, text
and availability on desktop.

This allows us to reduce the number of HTML elements and our CSS 
nesting significantly.

Also adjust spacing between elements to ensure that we are in line
with the design.

kasperg committed:
Update availability elements in search result element

Add more variation to the provided elements. This also resembles
actual usage more closely.

kasperg committed:
Remove default values from search result element component

It is not something that we use in other components. Defaults
come from Storybook anyway.

kasperbirch1 committed:
Replace align-items with align-content.

This makes sure to keep the flex spacing distance for accessibility

kasperbirch1 committed:
Change .scss name from IconFavourite to icon-favourite

spaceo commented on src/stories/icon/icon-favourite/IconFavourite.tsx:
```diff
@@ -0,0 +1,24 @@
+  // It is made as inline svg to be able to change fill value from props
+  return (
+    <svg
+      className="icon-favourite"
+      width="24"
+      height="24"
+      viewBox="0 0 24 24"
+      fill={fill ? "#000" : "none"}
```
Css is a weak spot of mine, but isn't it possible to define these attributes in css/sass instead?

spaceo commented on src/stories/search-result-element/SearchResultElement.tsx:
```diff
@@ -0,0 +1,78 @@
+      <div className="search-result-element__cover">
+        <Material
+          url={"images/book_cover_3.jpg"}
+          size={"small"}
+          animate={false}
+          tint={"120"}
```
A detail but you only need the {} with the `false` value

spaceo commented on src/stories/search-result-element/SearchResultElement.tsx:
```diff
@@ -0,0 +1,78 @@
+    </div>
+      <div className="search-result-element__availability">
+      <AvailabilityLabel
+          manifestation={"Bog"}
+          availability={"Hjemme"}
+          status={"available"}
+      />
+      <AvailabilityLabel
+          manifestation={"Ebog"}
+          availability={"Online"}
+          status={"available"}
+      />
+
+      <AvailabilityLabel
+          manifestation={"Lydbog (cd-mp3)"}
+          availability={"Udlånt"}
+          status={"unavailable"}
+      />
+      <AvailabilityLabel
+          manifestation={"Lydbog (net)"}
+          availability={"Online"}
+          status={"available"}
```
Same comment as the one above...

Adamik10 commented on src/stories/search-result-element/search-result-element.scss:
```diff
@@ -0,0 +1,45 @@
+  display: grid;
+  grid-template-columns: min-content 1fr;
+  gap: 16px;
+
```
Maybe a background color should be defined here? They say it's a variable that corresponds to Colors/Text/White in Figma.
![image](https://user-images.githubusercontent.com/28546954/171358092-2fcddf4f-b11d-4edc-8ea0-a5e0b0510798.png)
If we were to place the current search result element/item onto a colored background, the result would look something like zis:
![image](https://user-images.githubusercontent.com/28546954/171358593-21f1ab76-bfea-40ee-a5a2-f5c66b4e4730.png)

Adamik10 commented on base.scss:
```diff
@@ -26,6 +26,8 @@
 @import "./src/stories/counter/counter";
 @import "./src/stories/checkbox/checkbox";
 @import "./src/stories/availability-label/availability-label";
+@import "./src/stories/search-result-element/search-result-element";
```
Alrighty. I think search-result-element is a very vague name. I would suggest search-result-item / search-result-card / something more concrete. 
I also often call my branches something, make a new component, then code a bit, and realize that the name I gave it is not specific/good enough. Then it of course takes some time to rename the folders/files/references to files/class names and such, but I think it's worth it, and important to think about naming conventions as soon as possible in the process.

kasperbirch1 replied:
@kasperg @spaceo What is your opinion on that?

Adamik10 commented on src/stories/icon/icon-favourite/IconFavourite.stories.tsx:
```diff
@@ -0,0 +1,32 @@
+    design: {
+      type: "figma",
+      url:
+        "https://www.figma.com/file/Zx9GrkFA3l4ISvyZD2q0Qi/Designsystem?node-id=656%3A5407",
```
Nice that you include this link here 💯

Adamik10 commented on src/stories/icon/icon-favourite/IconFavourite.tsx:
```diff
@@ -0,0 +1,24 @@
+
+export const IconFavourite = ({ fill }: IconFavouriteProps) => {
+  // This svg is a copy from public/icons/basic/icon-heart.svg.
+  // It is made as inline svg to be able to change fill value from props
```
Great comment 🥇 loves it

Adamik10 replied:
Maybe we could also include one more line about what a future programmer working on this could do if they e.g. find this icon in the future, and it doesn't correspond to the original file, because the design has been changed. Something like:
// If you find out it no longer matches the original file: "src/xyz/icons/favoriteIcon.svg" please update it

Adamik10 commented on src/stories/icon/icon-favourite/IconFavourite.stories.tsx:
```diff
@@ -0,0 +1,32 @@
+  },
+  argTypes: {
+    fill: {
+      control: "boolean",
```
maybe we could also include default value here like in the SearchResultElement.stories.tsx? - for consistency. 
Or even just have a talk about whether we prefer one over the other, and how we use them. ❤️

Adamik10 commented on src/stories/search-result-element/search-result-element.scss:
```diff
@@ -0,0 +1,45 @@
+.search-result-element {
```
Have we thought about the hover state with this element? 
As far as I checked Figma, there is a drop shadow I can see, and also an arrow that should be shown on hover. I would imagine this could be a separate story for this card element. The original could be called something like "Default" and the other one "Hover". 
![image](https://user-images.githubusercontent.com/28546954/171376693-a06dcc8e-fac9-4078-8a60-b3112aef9dbf.png)

Adamik10 replied:
Also, we need to comply with the fact that all the cards should have the same height:
![image](https://user-images.githubusercontent.com/28546954/171380775-bae66b3f-c34f-4cc9-8033-4eb83edc48f9.png)

Currently, the card can be bloated and expand its intended height:
![image](https://user-images.githubusercontent.com/28546954/171381276-3434f9f8-cca2-412e-a932-b105f6eb8803.png)

Adamik10 commented on src/stories/search-result-element/search-result-element.scss:
```diff
@@ -0,0 +1,45 @@
+  gap: 10px;
+}
+
+.search-result-element__title {
```
The title should behave a bit differently according to Figma -> the title should break into a second line at some point, and afterwards cut off completely, as they say the cards should all have fixed height:
![image](https://user-images.githubusercontent.com/28546954/171379176-652499eb-9978-49a2-bb61-5e72ad7eb737.png)
This could happen with the current story:
![image](https://user-images.githubusercontent.com/28546954/171379728-fea7104c-2cc3-4535-ac87-f1b28255142e.png)

![image](https://user-images.githubusercontent.com/28546954/171377388-20aa4277-d66f-407f-9c27-d07b9c26df63.png)
I couldn't find out why in this this example they do not include availability label, but include a reserver button. 🤷  We should ask them about this as well tomorrow.

kasperbirch1 replied:
I have changed the width and spacing.

The reserve button has still not been added

kasperbirch1 replied:
Change figma link to https://www.figma.com/file/ETOZIfmgGS1HUfio57SOh7/S%C3%B8gning?node-id=4517%3A16671

Adamik10 commented on src/stories/search-result-element/search-result-element.scss:
```diff
@@ -0,0 +1,45 @@
+  }
+}
+
+.search-result-element__availability {
```
About availability - they mention two different cases: 
- one where there is one line of availability buttons:
![image](https://user-images.githubusercontent.com/28546954/171380158-9148d590-ba14-478b-b565-05808b17fbb1.png)

- and one where there are two:
![image](https://user-images.githubusercontent.com/28546954/171380007-21b925b2-77ec-4b7b-a152-3f16950583d6.png)

Also notice how they clearly all have the same alignment to left starting at 90px to the right of the headline:
![image](https://user-images.githubusercontent.com/28546954/171380462-9c99d911-4cb2-4ed4-bf1b-479264f83ab1.png)

Adamik10 reviewed (changes requested):
Beautiful work :) I'm impressed by how fast you mustered this, but I do think we need to match the design that Dept gave us for guidlines more closely. I left a bunch of comments all around. Feel free to start a conversation with me/the team about these things. 
❤️

kasperbirch1 committed:
Move favourite svg size control to css insted of inlinline values

kasperbirch1 committed:
Add comments

kasperbirch1 committed:
Move defaultValue to argTypes for consistency

kasperbirch1 committed:
Remove unnecessary jsx { }

kasperbirch1 committed:
Add white background color and hover shadow to search result element

kasperbirch1 committed:
Change width and spacing for search result element

Before, it did not fit with the design from figma

kasperbirch1 committed:
Add and show IconArrow on SearchResultElement hover

kasperbirch1 committed:
Change name from SearchResultElement to SearchResultItem

kasperbirch1 committed:
Add cursor pointer to search-result-item

kasperbirch1 committed:
set the same font size for meta text on search result item

kasperbirch1 committed:
Fix stylelint comments

kasperbirch1 committed:
Fix width for search-result-item__text
